ThunderTable/TableViewController+Collection.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  TableViewController+Collection.swift
+//  ThunderTable
+//
+//  Created by Simon Mitchell on 04/12/2019.
+//  Copyright © 2019 3SidedCube. All rights reserved.
+//
+
+import Foundation
+
+extension TableViewController {
+    
+    public subscript (index: IndexPath) -> (Section, Row)? {
+        
+        guard index.section < data.count else {
+            return nil
+        }
+        
+        let section = data[index.section]
+        guard index.row < section.rows.count else {
+            return nil
+        }
+        
+        return (section, section.rows[index.row])
+    }
+    
+    public subscript (row index: IndexPath) -> Row? {
+        
+        guard index.section < data.count else {
+            return nil
+        }
+        
+        let section = data[index.section]
+        guard index.row < section.rows.count else {
+            return nil
+        }
+        
+        return section.rows[index.row]
+    }
+    
+    public subscript (section index: IndexPath) -> Section? {
```

Comment:
I'd keep this one the same though as there's no need to waste time getting the row